### PR TITLE
Add internal/version and print version on boot

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

**Review by yolocs — APPROVED** [review]
👍

File: internal/version/version.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+)
+
+var (
+	// Name is the name of the binary. This can be overridden by the build
+	// process.
+	Name = "jvs"
+
+	// Version is the main package version. This can be overridden by the build
+	// process.
+	Version = "source"
```
Comment (sqin2019) [inline internal/version/version.go:29]: Hi, just wondering what are other Version values we are expecting other than "source", and when the Name and Version vars will be overridden?

Comment (sethvargo) [reply internal/version/version.go:29]: Hey @sqin2019 - they are overwritten by the build command: https://github.com/abcxyz/jvs/blob/3103e051920d7949d4b692b070032c72f43314de/.goreleaser.yaml#L38-L40

This allows us to inject the git tag and git commit directly into the build.

Comment (sqin2019) [reply internal/version/version.go:29]: Got it, thank you!

**Pull request merged**